phd-tests/framework/src/guest_os/windows_server_2022.rs

```diff
@@ -0,0 +1,67 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+//! Guest OS adaptations for Windows Server 2022 images. This adapter assumes
+//! the guest OS image has been generalized, that Cygwin has been installed, and
+//! that the local administrator user was configured at image generation time to
+//! have the appropriate password.
+
+use super::{CommandSequence, CommandSequenceEntry, GuestOs};
+
+pub(super) struct WindowsServer2022;
+
+impl GuestOs for WindowsServer2022 {
+    fn get_login_sequence(&self) -> CommandSequence {
+        CommandSequence(vec![
+            // The image is generalized and needs to be specialized. Let it boot
```

Comment:
out of curiosity: what does it mean for the image to be "generalized"/"specialized", exactly? and why does booting the image twice do what we need?